RuleDocumentation/UseIdenticalMandatoryParametersForDSC.md

```diff
@@ -4,13 +4,24 @@
 
 ## Description
 
-The `Get-TargetResource`, `Test-TargetResource` and `Set-TargetResource` functions of DSC Resource must have the same mandatory parameters.
+For script based DSC resources the `Get-TargetResource`, `Test-TargetResource` and `Set-TargetResource` functions must have identical mandatory parameters that are also the keys in the corresponding `mof` file.
```

Comment:
Oh no worries. 
So, I went through some of the script based resources and this is what I have found out:
- If a parameter is declared as `mandatory` in `get-targetresource`, then it has to be mandatory in the corresponding `set` and `test` functions.
- If a field is declared with attributes `Key` or `Required` in a mof file, then it should present as mandatory parameter in `get/set/test` functions. 
- `set` and `test` functions should have identical mandatory parameters.

Is that correct?